testsuite/features/step_definitions/command_steps.rb

```diff
@@ -23,16 +23,16 @@
   raise "reverse resolution for #{node.ip} returned #{result}, expected to see #{node.full_hostname}" unless result.include? node.full_hostname
 end
 
-Then(/^"([^"]*)" should communicate with the server$/) do |host|
+Then(/^"([^"]*)" should communicate with the server using "([^"]*)"/) do |host, ethernet|
   node = get_target(host)
-  node.run("ping -c1 #{$server.ip}")
-  $server.run("ping -c1 #{node.ip}")
+  node.run("ping -c 1 -I #{ethernet} #{$server.ip}")
```

Comment:
Is `-I #{ethernet}` really needed? I think we could just let the normal routing process decide which card is used.

But there is something really missing here. You should use `-4` to force IPv4.

Reply:
Yes, otherwise it was using the other ethernet, the public one, and the ping was successful.

Reply:
I think if we use -4 here it hides the problem if we enable ipv6 and this happens again. Not sure but I had the same impression at first that we needed to use -4. If we specify the IP and that's an IPv4 then that's taken care of automatically and we don't lose potential future problems.

Reply:
We don't address the problem for IPv6 now. In fact, we would need a ping for both protocols, IPv4 **and** IPv6.

But it makes little sense to test IPv6 before Ondrej addresses the issue SUSE/spacewalk#16487.

Reply:
I agree having two pings for IPv4 and IPv6, what it puzzles me is that we need to specify the card otherwise it connects using the public network. As shown above.